tests/integration/test_using_cached_dependencies.py

```diff
@@ -489,3 +497,32 @@ def update_main_repo(env_data, repo_dir, tmpdir, new_dep_commits, dep_repo):
             "FIRST_DEP_COMMIT": new_dep_commits[0],
             "SECOND_DEP_COMMIT": new_dep_commits[1],
         }
+    elif env_data["pkg_managers"] == ["rubygems"]:
+        with open(os.path.join(repo_dir, "Gemfile"), "a") as f:
+            f.write(f"gem 'tilt'\ngem 'my-package', git: '{env_data['dep_archive_baseurl']}'\n")
+        with open(os.path.join(repo_dir, "Gemfile.lock"), "r") as f:
+            gemlock = f.read()
+            gemlock.replace("zeitwerk (2.6.0)", "tilt (2.0.11)\n    zeitwerk (2.6.0)")
+            gemlock.replace("swagger-ui_rails!", "swagger-ui_rails!\n  tilt")
```

Comment:
`str.replace()` does not modify the string in place, you will have to do something like this
```suggestion
            gemlock = gemlock.replace("zeitwerk (2.6.0)", "tilt (2.0.11)\n    zeitwerk (2.6.0)")
```